studio/src/pages/[organizationSlug]/graph/[slug]/clients.tsx

```diff
@@ -0,0 +1,215 @@
+  TableRow,
+} from "@/components/ui/table";
+import {
+  Tooltip,
+  TooltipContent,
+  TooltipTrigger,
+} from "@/components/ui/tooltip";
+import { docsBaseURL } from "@/lib/constants";
+import { NextPageWithLayout } from "@/lib/page";
+import { cn } from "@/lib/utils";
+import { CommandLineIcon } from "@heroicons/react/24/outline";
+import { ExclamationTriangleIcon } from "@radix-ui/react-icons";
+import { useQuery } from "@tanstack/react-query";
+import { EnumStatusCode } from "@wundergraph/cosmo-connect/dist/common/common_pb";
+import { getClients } from "@wundergraph/cosmo-connect/dist/platform/v1/platform-PlatformService_connectquery";
+import { formatDistanceToNow } from "date-fns";
+import Link from "next/link";
+import { useRouter } from "next/router";
+import { BiAnalyse } from "react-icons/bi";
+import { IoBarcodeSharp } from "react-icons/io5";
+
+const ClientsPage: NextPageWithLayout = () => {
+  const router = useRouter();
+  const organizationSlug = router.query.organizationSlug as string;
+  const slug = router.query.slug as string;
+
+  const constructLink = (name: string, mode: "metrics" | "traces") => {
+    const filters = [];
+    const value = {
+      label: name,
+      value: name,
+      operator: 0,
+    };
+
+    const filter = {
+      id: "clientName",
+      value: [JSON.stringify(value)],
+    };
+    filters.push(filter);
+
+    if (mode === "metrics") {
+      return `/${organizationSlug}/graph/${slug}/analytics?filterState=${JSON.stringify(
+        filters,
+      )}`;
+    } else {
+      return `/${organizationSlug}/graph/${slug}/analytics/traces?filterState=${JSON.stringify(
+        filters,
+      )}`;
+    }
+  };
+
+  const { data, isLoading, error, refetch } = useQuery(
+    getClients.useQuery({
+      fedGraphName: slug,
+    }),
+  );
+
+  if (!data) return null;
+
+  if (!data || error || data.response?.code !== EnumStatusCode.OK)
+    return (
+      <EmptyState
+        icon={<ExclamationTriangleIcon />}
+        title="Could not retrieve changelog"
+        description={
+          data?.response?.details || error?.message || "Please try again"
+        }
+        actions={<Button onClick={() => refetch()}>Retry</Button>}
+      />
+    );
+
+  return (
+    <div className="flex flex-col gap-y-4">
+      {data.clients.length === 0 ? (
+        <EmptyState
+          icon={<CommandLineIcon />}
+          title="Push new operations to the registry using the CLI"
+          description={
+            <>
+              No clients found. Use the CLI tool to create one.{" "}
+              <a
+                target="_blank"
+                rel="noreferrer"
+                href={docsBaseURL + "/router/persisted-operations"}
+                className="text-primary"
+              >
+                Learn more.
+              </a>
+            </>
+          }
+          actions={
+            <CLI
+              command={`npx wgc operations push production -c <client-name> -f <path-to-file>`}
```

Comment:
This should be dynamic based on the active federated graph name.

Reply:
I've made it dynamic, PTAL